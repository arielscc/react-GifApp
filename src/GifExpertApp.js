import React, { useState } from 'react'
import AddCategory from './components/AddCategory'
import GifGrid from './components/GifGrid'

const GifExpertApp = () => {

  const [categories, setCategories] = useState(['goku'])

  return (
    <>
      <h2>Gift Expert App</h2>
      <AddCategory setCategories={ setCategories }/>
      <hr />
      {
        categories.map( category =>
          <GifGrid
            key = { category }
            category = { category } />
          )
      }
    </>
  )
}
export default GifExpertApp;