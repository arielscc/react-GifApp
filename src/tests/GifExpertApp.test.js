import React from 'react'
import GifExpertApp from "../GifExpertApp";
import { shallow } from "enzyme";
import '@testing-library/jest-dom'


describe('Test del componente <GiftExpertApp/>', () => {


  test('Probando el componente con el snapshot', () => {
    const wrapper = shallow( <GifExpertApp /> );
    expect( wrapper ).toMatchSnapshot();
  });

  test('debe mostrar una lista de categorias', () => {
    const categories = [ 'One punch', 'Dragon Ball'];
    const wrapper = shallow( <GifExpertApp defaultCategories = { categories }/> );

    expect( wrapper.find( 'GifGrid' ).length ).toBe( categories.length)
  });

});