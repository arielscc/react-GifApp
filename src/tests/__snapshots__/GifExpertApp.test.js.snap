// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test del componente <GiftExpertApp/> Probando el componente con el snapshot 1`] = `
<Fragment>
  <h2>
    Gift Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;
